src/components/SignUp/SignUpScreen.js

```diff
@@ -1,65 +1,282 @@
+  },
+  stepper: {
+    margin: 'auto',
+    width: '60%',
+  },
+  icon: {
+    color: 'red !important',
+  },
+});
+const BUTTONS = [
+  {
+    id: 0,
+    title: 'Buyer',
+    image: Fruit3,
+    fruit: {
+      marginTop: '10%',
+
+      height: '134px',
+      width: '98px',
+      left: '89px',
+      top: '46px',
+    },
+  },
+  {
+    id: 1,
+    title: 'Non-profit or Agency',
+    image: Fruit2,
+
+    fruit: {
+      marginTop: '10%',
+      height: '130px',
+      width: '105px',
+      left: '96px',
+      top: '50px',
+    },
+  },
+  {
+    id: 2,
+    title: 'Seller',
+    image: Fruit1,
+
+    fruit: {
+      marginTop: '10%',
+      height: '141px',
+      width: '91px',
+      left: '72px',
+      top: '39px',
+      // border-radius: '0px',
+    },
+  },
+];
 
 const INITIAL_FORM_STATE = {
-  username: '',
-  fullname: '',
+  firstName: '',
+  lastName: '',
+  email: '',
   password: '',
   confirmPassword: '',
+  org: '',
+  phone: '',
+  zipcode: '',
+  comments: '',
 };
 
+function getSteps() {
+  return ['Step 1', 'Step 2', 'Step 3', 'Confirmation'];
+}
+
+// TODO: Implement popup so when the user presses 'next' at step 3, there is a confirmation message
 export default function SignUpScreen() {
   const [formState, setFormState] = useState(INITIAL_FORM_STATE);
-  const [role, setRole] = React.useState('');
+  // Role mappings: 0 -> Buyer, 1 -> Agency, 2 -> Seller
+  const [role, setRole] = useState(-1);
   const [loading, setLoading] = useState(false);
   const [errorMsg, setErrorMsg] = useState('');
+  const [currentStep, setCurrentStep] = useState(1);
 
-  const roleChange = (event) => {
-    setRole(event.target.value);
-  };
-  const onSubmit = useCallback((event) => {
+  const classes = useStyles();
+
+  const handleChange = useCallback((event) => {
+    event.preventDefault();
+    setFormState(
+      {
+        ...formState,
+        [event.currentTarget.name]: event.currentTarget.value,
+      },
+    );
+  }, [formState, setFormState]);
+
+  const handleSubmit = useCallback((event) => {
     event.preventDefault();
     setLoading(true);
+    let res = null;
     try {
-      /* eslint-disable */
-      const res = signupUser(formState.username, formState.password, formState.fullname, role);
-
+      res = signupUser(formState.email, formState.password,
+        ROLE_NAMES[role],
+        formState.org, parseInt(formState.zipcode, 10), `${formState.firstName} ${formState.lastName}`,
+        formState.phone, formState.comments);
+      setErrorMsg('');
     } catch (err) {
+      console.log('error 1');
       if (err) {
+        console.log('error 2');
         setErrorMsg(err);
       }
-      
     }
-    setFormState(INITIAL_FORM_STATE);
-    setRole('');
-    setLoading(false);
-  }, [formState.username, formState.fullname, formState.password, role]);
+    console.log(res.result);
 
-  const onChange = useCallback((event) => {
+    if (res === null || errorMsg) {
```

Comment:
Ooooh cool! I figured this login stuff for selecting a different email! Refer to my slack messages to get that done and lmk if you have any questions